packages/entity-server/src/routes/hierarchy/middleware/filter.ts

```diff
@@ -129,14 +154,11 @@ export const extractFilterFromQuery = (
   }
 
   const filterClauses = queryFilter.split(CLAUSE_DELIMITER).map(toFilterClause);
-  const filter: Writable<NotNullValues<EntityFilter>> = {};
+  const filter: Writable<NotNullValues<Record<string, unknown>>> = {};
```

Comment:
@rohan-bes I had to make this type simpler because TS complained about the following in L161:

```ts
filter[formattedField] = convertValueToAdvancedCriteria(field, operator, value);
```

I think this is because we don't know if the converted value (`string | number`) can be assigned to the selected `field`, which would accept only `string` or only `number`. In the past everything was a `string`, so that was fine.

Perhaps more elaborate types could take care of this, but I wouldn't bother :stuck_out_tongue: